documentation/general/dash-sonic-hld.md

```diff
@@ -265,6 +266,73 @@ It is possible that a given packet can get a hit in route table and/or mapping t
 
   ![dash-outbound-meter](./images/dash-hld-outbound-meter-pipeline.png)
 
+## 2.5 FastPath
+
+This section captures the Sonic-Dash specifics of FastPath use-case. Detailed document on FastPath is captured here (![FastPath](https://github.com/sonic-net/DASH/blob/main/documentation/load-bal-service/load-balancer-v3.md))
+The following are the salient points and requirements:
+- FastPath kicks in when appliance receives an ICMP redirect that matches an existing unified flow
+- ICMP redirects are expected to be received from source and destination MUXes separately
+- Each ICMP redirect shall only update one side of the flow. (Src or Dst depending on the originating MUX)
+- ICMP packet is generated only for TCP. It is not generated for UDP traffic. 
+- FastPath example for Service Tunnels:
+	- After the first SYN pkt, appliance shall create two flows (one Outbound and another Inbound)
+	- Original Outbound packet shall have an inner IPv6 header and outer IPv4 (Src VIP-A and Dst VIP-B)
+	- After an ICMP redirect is received from VIP-B hosting MUX, the Outbound flow shall be fixed-up to have outer IPV4 dst address to use the Redirect IP of VIP-B. Same fixup for Inbound flow to change VIP-B to Redirect IP
+	- After an ICMP redirect is received from VIP-A hosting MUX, the Outbound flow shall be fixed-up to have outer IPV4 src address to use the Redirect IP of VIP-A. Same fixup for Inbound flow to change VIP-A to Redirect IP
+	- ICMP redirect shall have the original inner IPv6 address as the IP header's src and dst address. 
+	- Redirect info shall contain the transposed IPv6 address, src and dst ports, sequence number and the encap type (NVGRE in this case) in addition to redirect address. 
+		```
+			struct 
+			{
+		            uint32 Reserved;
+		            in6_addr Target;
+		            in6_addr Destination;
+		            uint8 Type;
+		            uint8 Length;
+		            uint8 Reserved2[6];            
+		            IPV6_HEADER Ipv6Header;
+		            uint16 SourcePort;
+		            uint16 DestinationPort;
+		            uint32 SequenceNumber;
+		        } Redirect;
+			
+			struct 
+			{ 
+			    uint32 Version; 
+			    uint16 AddrFamily; 
+			    uint16 EncapType; 
+			    uint32 EncapId; 
+			    union { 
+			        struct { 
+			            in_addr DipPAv4; 
+			            char VMMac[MAC_ADDR_SIZE]; 
+			        } Info4; 
+			        struct { 
+			            in6_addr DipPAv6; 
+			            char VMMac[MAC_ADDR_SIZE]; 
+			        } Info6; 
+		   	} Redirect_Info; 
+		```
```

Comment:
I feel some definition are missed or might not be clear enough:
- The meaning of certain fields, such as Type, EncapId.
- The value declaration of certain fields, such as EncapType.

Also it might be better to describe the packet in a complete way to make things more clear - starting from the outer most eth header.

Reply:
these fields are briefly described below. I think more details can be part of the detailed fastpath design

Reply:
I am not sure how the process usually looks like, but a bit wondering that should we wait for detailed design finished before merging it to HLD? this might help reduce confusions. Currently, if I wear the vendors hat and when I read the doc, I might be thinking I need to somehow handle the ICMP redirect packet in ASIC, which may or may not be true. I guess that might also be the reason people is asking why we have the C structs in the spec and what makes it relevant. But feel free to correct me, if I am wrong or missed anything.

Reply:
addressed as discussed, and clarified that detailed design can come as a separate PR later.